include/COLA.hh

```diff
@@ -195,36 +204,68 @@ namespace cola {
         void regGen(std::shared_ptr<VFactory>&& factory, const std::string& name){generatorMap.emplace(name, std::move(factory));}
         void regConv(std::shared_ptr<VFactory>&& factory, const std::string& name){converterMap.emplace(name, std::move(factory));}
         void regWrite(std::shared_ptr<VFactory>&& factory, const std::string& name){writerMap.emplace(name, std::move(factory));}
-    };
 
-    class ColaRunManager {
-    public:
-        explicit ColaRunManager(FilterAnsamble&& ansamble) : filterAnsamble(std::move(ansamble)) {}
-        ~ColaRunManager() = default;
-        void run();
-        void test();
-    private:
-        FilterAnsamble filterAnsamble;
+        MetaData parseStrToMeta(std::string data);
     };
 
 
-    inline cola::FilterAnsamble cola::MetaProcessor::parse(const cola::MetaData& data) {
+    inline cola::FilterAnsamble cola::MetaProcessor::parse(const cola::SimpleMetaData& data) {
+        FilterAnsamble ansamble;
+        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(data.generatorName)->create("")));
+        ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(data.converterName)->create(""))));
+        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(data.writerName)->create("")));
+        return ansamble;
+    }
+
+    inline cola::FilterAnsamble cola::MetaProcessor::parse(std::string data) {
         FilterAnsamble ansamble;
-        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(data.generatorName)->create()));
-        ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(data.converterName)->create())));
-        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(data.writerName)->create()));
+        MetaData meta = parseStrToMeta(data);
+        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(meta.generatorName)->create(meta.filterParamMap.at(meta.generatorName))));
```

Comment:
I think it is a good idea to move this part to `parse (const cola::MetaData&)` method and just call it here. This way exposing MetaData class to user actually serves some purpose and `parse(std::string)` becomes nice syntactic sugar.

Reply:
Also, maybe change signature to `parse(const std::string)`?

Reply:
you want to have it immutable by design?

Reply:
done